marine.c

```diff
@@ -227,13 +227,62 @@ static GHashTable *packet_filters;
 static int *packet_filter_keys[4096];
 static gboolean prefs_loaded = FALSE;
 
+static marine_pref_t optimize_perfs[] = {
+        //IPv4
+        "ip.summary_in_tree:false",
+        "ip.check_checksum:false",
+        "ip.use_geoip:false",
+
+        //IPv6
+        "ipv6.summary_in_tree:false",
+        "ipv6.use_geoip:false",
+
+        // TCP
+        "tcp.analyze_sequence_numbers:false",
+        "tcp.relative_sequence_numbers:false",
+        "tcp.track_bytes_in_flight:false",
+        "tcp.reassemble_out_of_order:false",
+        "tcp.check_checksum:false",
+        "tcp.tcp_summary_in_tree:false",
+        "tcp.display_process_info_from_ipfix:false",
+
+        // UDP
+        "udp.summary_in_tree:false",
+        "udp.process_info:false"
+
+};
+
+// Taken from prefs_set_pref_e (prefs.h)
+static const char* const pref_errors[] = {
+        "succeeded", "syntax error", "preference doesn't exist", "preference is obselete"
+};
 
 static void reset_epan_mem(capture_file *cf, epan_dissect_t *edt, gboolean tree, gboolean visual);
 
 inline static int is_only_bpf(const packet_filter* const filter) {
     return filter->has_bpf && filter->dfcode == NULL && filter->output_fields == NULL;
 }
 
+
+WS_DLL_PUBLIC int set_preferences(marine_pref_t* preferences, int num_of_prefs) {
```

Comment:
I think the API's output can be improved.
1. Printing to stderr isn't very useful. In Python it might be easy to catch it (which you don't currently do), but in other languages it might be much more difficult. Consider doing it like in `marine_add_filter` and writing the output to a buffer provided by the user.
2. The return value isn't very c-ish. I don't see how the amount of failed prefs is useful - I think a boolean is good enough with an error message that is easy to parse (array of strings? csv? what ever you like)